### ISSUE TITLE
Experimental feature failure: aws_ami

### ISSUE THREAD

**Issue by gtmtech** [hashicorp/terraform#9944]
terraform 0.7.9

Terraform resource in question

```
data "aws_ami" "citool" {
  most_recent = true
  executable_users = ["self"]
  filter {
    name = "tag:component"
    values = ["citool"]
  }
  filter {
    name = "tag:workflow_${environment["name"]}"
    values = ["published"]
  }
  owners = ["self"]
}
```

Failed to apply anyway with: 

```
Error applying plan:

2 error(s) occurred:

* data.aws_ami.appnode: InvalidParameterValue: The filter '' is invalid
	status code: 400, request id: ######
* data.aws_ami.citool: InvalidParameterValue: The filter '' is invalid
	status code: 400, request id: ######
```

Output during a plan:

```
Experimental feature failure! Please report a bug.

This is not an error. Your Terraform operation completed successfully.
Your real infrastructure is unaffected by this message.

While running, Terraform sometimes tests experimental features in the
background. These features cannot affect real state and never touch
real infrastructure. If the features work properly, you see nothing.
If the features fail, this message appears.

The following failures happened while running experimental features.
Please report a Terraform bug so that future Terraform versions that
enable these features can be improved!

You can report an issue at: https://github.com/hashicorp/terraform/issues

1 error(s) occurred:

* plan operation: ReadDataDiff "data.aws_ami.citool" had unequal configs (real, then shadow):

&terraform.ResourceConfig{ComputedKeys:[]string{"filter.name"}, Raw:map[string]interface {}{"owners":[]interface {}{"self"}, "most_recent":true, "executable_users":[]interface {}{"self"}, "filter":[]map[string]interface {}{map[string]interface {}{"name":"tag:component", "values":[]interface {}{"citool"}}, map[string]interface {}{"name":"tag:workflow_${environment[\"name\"]}", "values":[]interface {}{"published"}}}}, Config:map[string]interface {}{"filter":[]map[string]interface {}{map[string]interface {}{"values":[]interface {}{"citool"}, "name":"tag:component"}, map[string]interface {}{"values":[]interface {}{"published"}}}, "owners":[]interface {}{"self"}, "most_recent":true, "executable_users":[]interface {}{"self"}}, raw:(*config.RawConfig)(0xc421053da0)}

&terraform.ResourceConfig{ComputedKeys:[]string{"filter.1.name"}, Raw:map[string]interface {}{"most_recent":true, "executable_users":[]interface {}{"self"}, "filter":[]map[string]interface {}{map[string]interface {}{"name":"tag:component", "values":[]interface {}{"citool"}}, map[string]interface {}{"name":"tag:workflow_${environment[\"name\"]}", "values":[]interface {}{"published"}}}, "owners":[]interface {}{"self"}}, Config:map[string]interface {}{"owners":[]interface {}{"self"}, "most_recent":true, "executable_users":[]interface {}{"self"}, "filter":[]map[string]interface {}{map[string]interface {}{"name":"tag:component", "values":[]interface {}{"citool"}}, map[string]interface {}{"values":[]interface {}{"published"}}}}, raw:(*config.RawConfig)(0xc420829860)}

This is not an error. Your terraform operation completed successfully
and your real infrastructure is unaffected by this message.
```

**Comment by kwilczynski** [comment]
Related #9945.

**Comment by mitchellh** [comment]
Thanks! This appears to be fixed on master. I honestly don't know which PR fixed this but I can reproduce it with the v0.7.9 tag but not master.
